### PR TITLE
Support tooltip for annotations

### PR DESCRIPTION
**Pull request by PatrickTasse**
Move the fetch tooltip implementation to the TspDataProvider class.
Provide a method to fetch state tooltip and a method to fetch annotation
tooltip. Create the requested element for the request from the element
model.

Add the category to the annotation model when fetching annotations.

Adapt the tooltip content according to the type of element.

Signed-off-by: Patrick Tasse <patrick.tasse@ericsson.com>

### CODE REVIEW

**Review by bhufmann — COMMENTED** [review]
@PatrickTasse  Looks good to me. I tested using all dependent changes in timeline-chart, tsp-typescript-client and trace server. I'll approve it once all dependencies are merged and we have a successful build here

**Review by MatthewKhouzam — APPROVED** [review]
LGTM, thanks.

**Comment by bhufmann** [comment]
@PatrickTasse I'm going to merge it since the Trace Compass server has been updated for that.

**Pull request merged**
